File: old-rust-tests/tests/dtsh_run_basic.rs
mod dt_test_utils;
use dt_test_utils::dt_run_file;

#[test]
pub fn say_hello() {
    let res = dt_run_file("tests/basic.dt");

    assert_eq!("", res.stderr);
    assert!(res.status.success());
    assert_eq!("Hello world!\n", res.stdout);
}

File: src/tests/dtsh_run_basic.zig
const std = @import("std");
const expectEqual = std.testing.expectEqual;
const expectEqualStrings = std.testing.expectEqualStrings;

const dtRunFile = @import("dt_test_utils.zig").dtRunFile;

test "say_hello" {
    const res = try dtRunFile("./src/tests/basic.dt");
    try expectEqualStrings("", res.stderr);
    try expectEqual(@as(u8, 0), res.term.Exited);
    try expectEqualStrings("Hello world!\n", res.stdout);
}
